hedera-node/hedera-addressbook-service-impl/src/test/java/com/hedera/node/app/service/addressbook/impl/test/CertificatePemTest.java

```diff
@@ -81,9 +81,35 @@ void invalidCertificatePem() throws CertificateException, IOException {
     void invalidBytesInPemCannotRead() throws IOException {
         final var genPemPath = Path.of(tmpDir.getPath() + "/generated.pem");
         writeCertificatePemFile(genPemPath, Bytes.wrap("anyString").toByteArray());
-        final var exception = assertThrows(IOException.class, () -> readCertificatePemFile(genPemPath));
-        assertThat(exception.getMessage()).contains("problem parsing cert: java.io.EOFException:");
+        final var exception = assertThrows(CertificateException.class, () -> readCertificatePemFile(genPemPath));
+        assertThat(exception.getMessage()).contains("Can not read the certificate from the file ");
         final var msg = assertThrows(PreCheckException.class, () -> validateX509Certificate(Bytes.wrap("anyString")));
         assertEquals(ResponseCodeEnum.INVALID_GOSSIP_CA_CERTIFICATE, msg.responseCode());
     }
+
+    @Test
+    void badX509CertificateFailedOnReadAndValidation() throws IOException {
+        final var pemFileName = "badX509certificate.pem";
+        final var pemFilePath = loadResourceFile(pemFileName);
+        final var exception = assertThrows(CertificateException.class, () -> readCertificatePemFile(pemFilePath));
+        assertEquals("Can not read the certificate from the file badX509certificate.pem", exception.getMessage());
+
+        final byte[] certBytes = Files.readAllBytes(pemFilePath);
+        final var msg = assertThrows(PreCheckException.class, () -> validateX509Certificate(Bytes.wrap(certBytes)));
+        assertEquals(ResponseCodeEnum.INVALID_GOSSIP_CA_CERTIFICATE, msg.responseCode());
+    }
```

Comment:
May I suggest one more test?
```suggestion
    void badX509CertificateFailedOnReadAndValidation() throws IOException {
        final var pemFileName = "goodX509certificate.pem";
        final var pemFilePath = loadResourceFile(pemFileName);
        final var exception = assertThrows(CertificateException.class, () -> readCertificatePemFile(pemFilePath));
        assertEquals("Can not read the certificate from the file goodX509certificate.pem", exception.getMessage());

        final byte[] certBytes = Files.readAllBytes(pemFilePath);
        final var msg = assertThrows(PreCheckException.class, () -> validateX509Certificate(Bytes.wrap(certBytes)));
        assertEquals(ResponseCodeEnum.INVALID_GOSSIP_CA_CERTIFICATE, msg.responseCode());
    }
```

This tests if we are sent a good certificate in pem format instead of binary DER.

Reply:
The above test failed because this line of code`final var exception = assertThrows(CertificateException.class, () -> readCertificatePemFile(pemFilePath)); `. In goodX509CertificateSuccessOnReadAndValidation(), we tested readCertificatePemFile() and does not throw.

Reply:
Sorry, wrote that in a hurry and missed a line or two.
I've updated the suggested test.

Reply:
Re-ran your new code, the test still failed on line assertEquals(ResponseCodeEnum.INVALID_GOSSIP_CA_CERTIFICATE, msg.responseCode());. In goodX509CertificateSuccessOnReadAndValidation(), we tested assertDoesNotThrow(() -> validateX509Certificate(Bytes.wrap(certBytes)));

Reply:
I suspected this line might fail.  This is the scenario that _should_ fail validation but does not.
This test demonstrates that the validation needs to be improved.